core/src/main/java/hivemall/topicmodel/OnlineLDAModel.java

```diff
@@ -0,0 +1,530 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package hivemall.topicmodel;
+
+import hivemall.annotations.VisibleForTesting;
+import hivemall.model.FeatureValue;
+import hivemall.utils.lang.ArrayUtils;
+import hivemall.utils.math.MathUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+
+import org.apache.commons.math3.distribution.GammaDistribution;
+import org.apache.commons.math3.special.Gamma;
+
+public final class OnlineLDAModel {
+
+    // number of topics
+    private int _K;
+
+    // prior on weight vectors "theta ~ Dir(alpha_)"
+    private float _alpha = 1 / 2.f;
+
+    // prior on topics "beta"
+    private float _eta = 1 / 20.f;
+
+    // total number of documents
+    // in the truly online setting, this can be an estimate of the maximum number of documents that could ever seen
+    private long _D = -1L;
+
+    // defined by (tau0 + updateCount)^(-kappa_)
+    // controls how much old lambda is forgotten
+    private double _rhot;
+
+    // positive value which downweights early iterations
+    @Nonnegative
+    private double _tau0 = 1020;
+
+    // exponential decay rate (i.e., learning rate) which must be in (0.5, 1] to guarantee convergence
+    private double _kappa = 0.7d;
```

Comment:
`_tau0` and `_kappa` should be fixed as `final` in the constructor for readability.

Check other constructor parameters as well.